### PR TITLE
Completely remove the RT sections of this crate

### PR DESCRIPTION
**Pull request by MabezDev**
Expose atomic emulation parts only.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -82,17 +13,25 @@ macro_rules! amo {
     };
 }
 
-pub unsafe fn atomic_emulation(frame: &mut TrapFrame) -> bool {
-    static mut S_LR_ADDR: usize = 0;
+/// is_atomic_instruction
+/// 
+/// Take the program counter address and returns whether the instruction at that address is an atomic one
+pub unsafe fn is_atomic_instruction(pc: usize) -> bool {
+    (*(pc as *const usize) & 0b1111111) != 0b0101111
```
Comment (bjoernQ) [inline src/lib.rs]: should be `==` not `!=`

Comment (MabezDev) [reply src/lib.rs]: Oops, thanks for finding that! Fixed.

**Review by bjoernQ — APPROVED** [review]
LGTM

Works fine in my tests

**Comment by MabezDev** [comment]
Released as `0.4`.

**Pull request merged**